packages/components/src/internal/components/notifications/ServerNotifications.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { ReactNode } from 'react';
+import { DropdownButton } from 'react-bootstrap';
+
+import { getPipelineActivityData, markAllNotificationsAsRead, markNotificationsAsRead } from './actions';
+import { ServerActivity } from './model';
+import { ServerActivityList } from './ServerActivityList';
+import { LoadingSpinner } from '../../../index';
+
+interface Props {
+    maxListingSize?: number;
+}
+
+interface State {
+    serverActivity: ServerActivity;
+    isLoading: boolean;
+    error: string;
+    show: boolean;
+}
+
+export class ServerNotifications extends React.Component<Props, State> {
+    static defaultProps = {
+        maxListingSize: 8,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            serverActivity: undefined,
+            isLoading: true,
+            error: undefined,
+            show: false,
+        };
+    }
+
+    componentDidMount(): void {
+        getPipelineActivityData(this.props.maxListingSize)
+            .then(response => {
+                this.setState(() => ({ serverActivity: response, isLoading: false }));
+            })
+            .catch(reason => {
+                this.setState(() => ({ error: reason, isLoading: false }));
+            });
+    }
+
+    markAllRead = (): void => {
+        const { data } = this.state.serverActivity;
+        const now = new Date().toTimeString();
+        const updatedData = [];
+        data.forEach(activity => {
+            if (activity.RowId) {
+                updatedData.push(activity.mutate({ ReadOn: now }));
+            } else {
+                updatedData.push(activity);
+            }
+        });
+        markAllNotificationsAsRead()
+            .then(() => {
+                this.setState(state => (
+                    { serverActivity: Object.assign({}, state.serverActivity, { data: updatedData, unreadCount: 0 })}
+                ));
+            })
+            .catch(() => {
+                    console.error('Unable to mark all notifications as read');
+            });
+    };
+
+    onRead = (id: number): void => {
+        markNotificationsAsRead([id])
+            .then(() => {
+                this.setState(state => {
+                    const dataIndex = state.serverActivity.data.findIndex(d => d.RowId === id);
```

Comment:
I image this might change altogether when we've wired up the redux store, and yes I imagine it will result in re-querying for the data after such a change, so I'm going to leave it as is for the time being.